src/components/StatusBadge/index.js

```diff
@@ -0,0 +1,110 @@
+import React, { Component } from "react";
+import PropTypes from "prop-types";
+import styled from "styled-components";
+import Badge, { StyledBadge } from "./Badge";
+import Tooltip from "../Tooltip";
+import { themes } from "../../theme";
+
+import { badgeTypes, badgeProps } from "./constants";
+import { directions } from "../Tooltip/constants";
+
+const StatusBadgeCotainer = styled.div`
+  display: flex;
+  .badge {
+    margin-left: 8px;
+
+    &:first-child,
+    &.hidden {
+      margin-left: 0px;
+    }
+
+    &.hidden {
+      margin-top: 8px;
+      :first-child {
+        margin-top: 0;
+      }
+    }
+  }
+`;
+
+const HiddenBadgesCount = styled(StyledBadge)`
+  background: ${themes.global.transparent};
+  color: ${themes.global.onyx.light};
+`;
+
+const BadgeTooltip = styled(Tooltip)`
+  padding: 8px;
+`;
+
+class StatusBadge extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      isOpened: false
+    };
+  }
+
+  mouseLeave = () => {
+    this.setState({
+      isOpened: false
+    });
+  };
+
+  elementHovered = e => {
+    const data = Tooltip.getDimensionsFromEvent(e);
+    this.setState({
+      isOpened: true,
+      ...data
+    });
+  };
+
+  render() {
+    const { visibleBadges, hiddenBadges, tooltipDirection, type } = this.props;
+    const { isOpened, ...position } = this.state;
+
+    return (
+      <StatusBadgeCotainer>
+        {visibleBadges.map(item => (
+          <Badge {...item} type={type} key={item.label} />
```

Comment:
I think we should verify that `item` is an object before trying to access any properties

Reply:
visibleBadges is defined as arrayOf objects in propTypes. Isn't that enough? The developer will get warning if tries to pass something different.

Reply:
Ok, that's fine